org.mixedrealitytoolkit.uxcore/Experimental/NonNativeKeyboard/NonNativeKeyTouchAdapter.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Mixed Reality Toolkit Contributors
+// Licensed under the BSD 3-Clause
+using System.Collections;
+using MixedReality.Toolkit;
+using MixedReality.Toolkit.Input;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.XR.Interaction.Toolkit;
+
+namespace MixedReality.Toolkit.UX.Experimental
+{
+    public class NonNativeKeyTouchAdapter : MonoBehaviour
+    {
+        private const float ColliderMargin = 30.0f;
+        private const float ColliderThickness = 70.0f;
+        private const float ColliderZDelta = 20.0f;
+        private const float ReClickDelayTime = 1.0f;
+        private const float AnimationTime = 0.2f;
+        private const float AnimationMovementDelta = 20.0f;
+
+        private StatefulInteractable interactable;
+        private Graphic image;
+        private float lastClickTime;
+        private bool isInitialized;
+        private Vector3 defaultPosition;
+        private Vector3 animatedPosition;
+        private BoxCollider buttonCollider;
+        private Vector3 buttonColliderDefaultCenter;
+        private Color defaultImageColor;
+
+        private void Awake()
+        {
+            defaultPosition = transform.localPosition;
+            animatedPosition = defaultPosition + new Vector3(0, 0, AnimationMovementDelta);
+        }
+
+        private void OnEnable()
+        {
+            transform.localPosition = defaultPosition;
+            lastClickTime = Time.time;
+            if (isInitialized)
+            {
+                 buttonCollider.center = buttonColliderDefaultCenter;
+            }
+
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            if (isInitialized)
+            {
+                return;
+            }
+            isInitialized = true;
+
+            var rectTransform = GetComponent<RectTransform>();
```

Comment:
I prefer having the caller explicitly set the component values via serialized fields, instead of using `GetComponent`.    Doing so will allow this component to be utilized by various other hierarchies.  This would also remove the need for "required components" on this game object.

Curious what other reviewers think  (@keveleigh ?)

Reply:
The whole idea is to create an add-on for the existing keyboard. This behaviour is added in code (by NonNativeKeyboardTouchAdapter) and isn't actually indented to be used on it's own

Reply:
The only way to get this done is to add, this behaviour, manually, one by one to _every_ key in the keyboard - as the keys are no prefabs. I feel like a broken record repeating this ;)